x86 Tests Failing

#### **Describe the bug**
```
2: ======================================================================
2: FAIL: test_run (github.test_CVE_2017_11591.TestCvePoC)
2: ----------------------------------------------------------------------
2: Traceback (most recent call last):
2:   File "/exiv2/tests/system_tests.py", line 652, in test_run
2:     self.compare_stderr(i, command, processed_stderr, stderr)
2:   File "/exiv2/tests/system_tests.py", line 966, in check_no_ASAN_UBSAN_errors
2:     self.assertNotIn(UBSAN_MSG, got_stderr)
2: AssertionError: 'runtime error' unexpectedly found in 'Error: Directory Image, entry 0x0000 has invalid size 4286578688*8; skipping entry.\nWarning: Directory Image, entry 0x0111: Strip 0 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 1 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 2 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 3 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 4 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 5 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 7 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 8 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 10 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 12 is outside of the data area; ignored.\n/exiv2/src/tiffcomposite_int.cpp:298:45: runtime error: pointer index expression with base 0xf2135000 overflowed to 0x53135000\n'
2: 
2: ======================================================================
2: FAIL: test_run (github.test_CVE_2017_14859.TestCvePoC)
2: ----------------------------------------------------------------------
2: Traceback (most recent call last):
2:   File "/exiv2/tests/system_tests.py", line 652, in test_run
2:     self.compare_stderr(i, command, processed_stderr, stderr)
2:   File "/exiv2/tests/bugfixes/github/test_CVE_2017_14859.py", line 19, in compare_stderr
2:     self.assertIn(expected_stderr, got_stderr)
2: AssertionError: 'Exiv2 exception in print action for file /exiv2/test/data/005-invalid-mem:\ncorrupted image metadata\n' not found in 'Warning: Directory Image, entry 0x011a has unknown Exif (TIFF) type 64772; setting type size 1.\n/exiv2/src/tiffvisitor_int.cpp:1319:48: runtime error: pointer index expression with base 0xef802138 overflowed to 0x2f802140\n'
2: 
2: ----------------------------------------------------------------------
2: Ran 300 tests in 56.875s
2: 
2: FAILED (failures=2, skipped=10)

```
#### **To Reproduce**
Dockerfile:
```
FROM i386/debian

RUN apt update; apt install -y cmake g++ clang make libexpat1-dev zlib1g-dev libbrotli-dev libssh-dev libcurl4-openssl-dev libgtest-dev libxml2-utils libinih-dev python3-dev git gettext libgmock-dev libgtest-dev neovim vim nano && \
  git clone https://github.com/Exiv2/exiv2 && cd exiv2 && \
  cmake . -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DEXIV2_ENABLE_NLS=TRUE -DEXIV2_BUILD_SAMPLES=TRUE -DEXIV2_ENABLE_WEBREADY=ON -DEXIV2_ENABLE_CURL=ON -DEXIV2_ENABLE_VIDEO=ON -DEXIV2_BUILD_SAMPLES=ON -DEXIV2_BUILD_UNIT_TESTS=ON -DEXIV2_TEAM_USE_SANITIZERS=ON -DEXIV2_TEAM_EXTRA_WARNINGS=ON && \
  cmake --build . --parallel && \
  ctest . -VV
```

```
docker build -t test-image -f Dockerfile
```

#### **Expected behavior**
Tests passing

ping @kevinbackhouse

errors on arm32:

```
======================================================================
FAIL: test_run (github.test_CVE_2017_11591.TestCvePoC)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 652, in test_run
    self.compare_stderr(i, command, processed_stderr, stderr)
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 966, in check_no_ASAN_UBSAN_errors
    self.assertNotIn(UBSAN_MSG, got_stderr)
AssertionError: 'runtime error' unexpectedly found in 'Error: Directory Image, entry 0x0000 has invalid size 4286578688*8; skipping entry.\nWarning: Directory Image, entry 0x0111: Strip 0 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 1 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 2 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 3 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 4 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 5 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 7 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 8 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 10 is outside of the data area; ignored.\nWarning: Directory Image, entry 0x0111: Strip 12 is outside of the data area; ignored.\n/home/mangix/devstuff/exiv2/src/tiffcomposite_int.cpp:297:45: runtime error: pointer index expression with base 0xb1d51000 overflowed to 0x12d51000\n'

======================================================================
FAIL: test_run (github.test_CVE_2018_12265.AdditionOverflowInLoaderExifJpeg)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 652, in test_run
    self.compare_stderr(i, command, processed_stderr, stderr)
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 773, in compare_stderr
    self._compare_output(
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 745, in _compare_output
    self.assertMultiLineEqual(
AssertionError: 'Erro[319 chars]ata area; ignored.\n' != 'Erro[319 chars]ata area; ignored.\nUncaught exception: Overflow in addition\n'
  Error: Upper boundary of data for directory Image, entry 0x00fe is out of bounds: Offset = 0x0000002a, size = 64, exceeds buffer size by 22 Bytes; truncating the entry
  Warning: Directory Image, entry 0x0201: Strip 0 is outside of the data area; ignored.
  Warning: Directory Image, entry 0x0201: Strip 7 is outside of the data area; ignored.
+ Uncaught exception: Overflow in addition
 : Standard error does not match

======================================================================
FAIL: test_run (github.test_pr_2244.test_pr_2244)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 652, in test_run
    self.compare_stderr(i, command, processed_stderr, stderr)
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 773, in compare_stderr
    self._compare_output(
  File "/home/mangix/devstuff/exiv2/tests/system_tests.py", line 745, in _compare_output
    self.assertMultiLineEqual(
AssertionError: 'Unca[35 chars] high=========================================[2458 chars]NG\n' != 'Unca[35 chars] high\n'
+ Uncaught exception: year adjustment too high
- Uncaught exception: year adjustment too high=================================================================
- ==28940==ERROR: AddressSanitizer: stack-use-after-scope on address 0xbebc5328 at pc 0xb69caadb bp 0xbebc4ee8 sp 0xbebc4ef0
- READ of size 256 at 0xbebc5328 thread T0
-     #0 0xb69caad8 in __interceptor_memcpy ../../../../src/libsanitizer/sanitizer_common/sanitizer_common_interceptors.inc:806
-     #1 0xb449c770 in std::ctype<char>::_M_widen_init() const (/lib/arm-linux-gnueabihf/libstdc++.so.6+0x95770)
-     #2 0x58e1c4 in std::basic_ostream<char, std::char_traits<char> >& std::endl<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&) [clone .isra.0] (/home/mangix/devstuff/exiv2/bb/bin/exiv2+0x9e1c4)
-     #3 0x58c492 in main (/home/mangix/devstuff/exiv2/bb/bin/exiv2+0x9c492)
-     #4 0xb3dda59c in __libc_start_main (/lib/arm-linux-gnueabihf/libc.so.6+0x1759c)
-
- Address 0xbebc5328 is located in stack of thread T0
- SUMMARY: AddressSanitizer: stack-use-after-scope ../../../../src/libsanitizer/sanitizer_common/sanitizer_common_interceptors.inc:806 in __interceptor_memcpy
- Shadow bytes around the buggy address:
-   0x37d78a10: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
-   0x37d78a20: 00 00 00 00 00 00 00 00 00 00 f8 00 00 00 00 00
-   0x37d78a30: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
-   0x37d78a40: 00 00 f8 00 00 00 00 00 00 00 00 00 f8 00 00 00
-   0x37d78a50: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
- =>0x37d78a60: 00 00 00 00 f8[f8]f8 00 00 00 00 00 00 00 00 00
-   0x37d78a70: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
-   0x37d78a80: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
-   0x37d78a90: 00 00 00 00 00 00 00 00 00 00 00 00 f8 00 00 00
-   0x37d78aa0: f8 00 f8 00 00 00 00 00 00 00 00 00 00 00 00 00
-   0x37d78ab0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
- Shadow byte legend (one shadow byte represents 8 application bytes):
-   Addressable:           00
-   Partially addressable: 01 02 03 04 05 06 07
-   Heap left redzone:       fa
-   Freed heap region:       fd
-   Stack left redzone:      f1
-   Stack mid redzone:       f2
-   Stack right redzone:     f3
-   Stack after return:      f5
-   Stack use after scope:   f8
-   Global redzone:          f9
-   Global init order:       f6
-   Poisoned by user:        f7
-   Container overflow:      fc
-   Array cookie:            ac
-   Intra object redzone:    bb
-   ASan internal:           fe
-   Left alloca redzone:     ca
-   Right alloca redzone:    cb
-   Shadow gap:              cc
- ==28940==ABORTING
 : Standard error does not match

```

--- a/src/tiffcomposite_int.cpp
+++ b/src/tiffcomposite_int.cpp
@@ -295,7 +295,6 @@ void TiffImageEntry::setStrips(const Value* pSize, const byte* pData, size_t siz
   }
   for (size_t i = 0; i < pValue()->count(); ++i) {
     const size_t offset = pValue()->toUint32(i);
-    const byte* pStrip = pData + baseOffset + offset;
     const size_t size = pSize->toUint32(i);
 
     if (size > sizeData || offset > sizeData - size || baseOffset > sizeData - size - offset) {
@@ -304,6 +303,7 @@ void TiffImageEntry::setStrips(const Value* pSize, const byte* pData, size_t siz
                   << tag() << ": Strip " << std::dec << i << " is outside of the data area; ignored.\n";
 #endif
     } else if (size != 0) {
+      const byte* pStrip = pData + baseOffset + offset;
       strips_.emplace_back(pStrip, size);
     }
   }

--- a/src/tiffvisitor_int.cpp
+++ b/src/tiffvisitor_int.cpp
@@ -1316,7 +1316,7 @@ void TiffReader::readTiffEntry(TiffEntryBase* object) {
                   << size
                   << ", exceeds buffer size by "
                   // cast to make MSVC happy
-                  << static_cast<size_t>(pData + size - pLast_) << " Bytes; truncating the entry\n";
+                  << size - static_cast<size_t>(pLast_ - pData) << " Bytes; truncating the entry\n";
 #endif
         size = 0;
       }

